opennms-wmi/src/main/java/org/opennms/netmgt/collectd/WmiCollector.java

```diff
@@ -265,8 +273,8 @@ private boolean isGroupAvailable(final WmiAgentState agentState, final Wpm wpm)
         return true;
     }
 
-    private Resource getWmiResource(CollectionAgent agent, String resourceType, NodeLevelResource nodeResource, String instance) {
-        ResourceType rt = DataCollectionConfigFactory.getInstance().getConfiguredResourceTypes().get(resourceType);
+    private Resource getWmiResource(final Map<String, ResourceType> resourceTypesMap, final CollectionAgent agent, final String resourceType, final NodeLevelResource nodeResource, final String instance) {
+        ResourceType rt = resourceTypesMap.get(resourceType);
```

Comment:
Can use the `DeferredGenericTypeResource` which doesn't require access to the actual definition at collection time - see https://github.com/OpenNMS/opennms/blob/develop/features/jdbc-collector/src/main/java/org/opennms/netmgt/collectd/JdbcCollector.java#L175